### ISSUE TITLE
PPO Complex Obs/Action Space

### ISSUE THREAD

**Issue by ttumiel** [vwxyzjn/cleanrl#353]
## Problem Description
Would it be useful to add a complex (nested/dictionary) action and obs space variant of the PPO algo? I did this for `minerl` and wondered if it would be useful to contribute into the main library? I'd happily make a PR.

## Checklist
- [x] I have checked that there is no similar [issue](https://github.com/vwxyzjn/cleanrl/issues) in the repo.
- [x] I have checked the [documentation site](https://docs.cleanrl.dev/) and found not relevant information in [GitHub issues](https://github.com/vwxyzjn/cleanrl/issues).

## Current Behavior
Currently PPO only supports continuous or discrete actions separately and a single array observation.

## Expected Behavior
PPO can support arbitrary complex action and observation spaces.

## Possible Solution
- Use `tree` to map over actions and observation. 
- Store arrays in the same struct shape as the obs space or flatten them for storage and unflatten when passing to the network.

**Comment by vwxyzjn** [comment]
Yes that would be great. I suggest implementing it based on #338. #338 uses EnvPool's async API, which is equivalent to the regular vec env when `async_batch_size = num_envs`. 

I was thinking about this issue more and think that you should have two types of observations:

1. Vector obs
2. Image obs

And for these two obs types we need to pair it with corresponding networks. 

Feel free to make a PR :) Thanks.

Cc @edbeeching, this PR could help deal with Godot rl environments.

**Comment by vwxyzjn** [comment]
Hi @ttumiel just following up with this. Are you still interested in the issue?

**Comment by ttumiel** [comment]
Yes! Sorry about the delay, I'll post a PR soon :)
